src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -265,12 +265,29 @@ must be supplied as the value to be matched."
         (str "ALTER TABLE " table " ALTER COLUMN " column
              " RESTART WITH " restartid))))))
 
-(pls/defn-validated clj->pgjson :- PGobject
-  "Convert a clojure object to a json PGobject"
-  [value :- (s/maybe (s/either {s/Any s/Any} [{s/Any s/Any}]))]
-  (doto (PGobject.)
-    (.setType "json")
-    (.setValue (json/generate-string value))))
+(defn sql-hash-as-str
+  [column]
+  (if (postgres?)
+    (format "trim(leading '\\x' from %s::text)" column)
+    column))
+
+(defn sql-uuid-as-str
+  [column]
+  (if (postgres?)
+    (format "%s::text" column)
```

Comment:
Incidentally, we already have scast in honeysql.clj which does the same thing more or less, but uses cast(col as text) instead.